shared_sql_utils/src/dsn.rs

```diff
@@ -249,6 +249,7 @@ impl<'a> DSNIterator<'a> {
                 ("Password", &dsn_opts.password),
                 ("Uri", &dsn_opts.uri),
                 ("User", &dsn_opts.user),
+                ("LogLevel", &dsn_opts.log_level),
```

Comment:
Did you consider removing the log level from the DSN configuration GUI instead?

I didn't realize/forgot we have a field for log level in the DSN config GUI.
The only reason I didn't remove it from the URI support is because I don't want to break older PowerBI or even ODBC driver installation.
However, I am not a big fan of DSN/connection level configuration which are overriding other connections properties. It makes things hard to document and frustrating for users.
Also, since we would only allow Driver level logging for PowerBI, it might be better to standardize this.
Why was setting the driver log level at the Driver level not enough?

Reply:
I had not considered removing it, just noticed this was not working when doing compatibility testing so checked out out and found it was a simple fix. If it would be more sustainable / a better long term solution however, I'd be happy to do what you are suggesting

Reply:
I'm curious, what tools support multiple different simultaneous DSN configurations? If this is a thing, we may want to support adding the dsn name as a key in the log.

Reply:
I'd expect most tools support multiple connections. ETL or SQL IDEs will for sure.  I haven't tested the tools. You guys should have a better idea for the tools we tested. Generally, if the tools also you to have multiple source for your report, then it has to allow multiple connection.

We have [SQL-1795](https://jira.mongodb.org/browse/SQL-1795) for supporting connection log level properly. I think it might be better to just tackle it at that time.

Side question for @terakilobyte : What would add the dsn key in the log do? How/where do you think about using it? Do you mean logging the DSN name for each line if we connected through a DSN?

Reply:
@nbagnard Yeah that's my thought, though the limited subset of tools I tested didn't have simultaneous connections. However, if this is common, it may be something to think about.